apps/server/src/services/rundown-service/RundownService.ts

```diff
@@ -164,29 +165,29 @@ export async function addEvent(eventData: Partial<OntimeEvent> | Partial<OntimeD
   let newEvent: Partial<OntimeBaseEvent> = {};
   const id = generateId();
 
-  // TODO: filter the parameters that exist in the event, use the parserUtils
+  let insertIndex = 0;
+  if (typeof eventData?.after !== 'undefined') {
```

Comment:
Why not `if (eventData?.after === undefined)`?

Reply:
generally checking with typeof is safer as it also covers the case where the value is undeclared

eg:
```
typeof doesNotExist === "undefined" // true
doesNotExist === undefined // ReferenceError: doesNotExist is not defined
```

I guess the better approach would be to evaluate this case by case, but I tend to use the safer one unless I am very concerned with the performance of a function and very certain that it will always be declared

Reply:
Seems to be having the same effect in the browser for non-existent object properties. Just pointing it out. Feel free to resolve!

<img width="326" alt="Screenshot 2023-08-16 at 3 08 55 PM" src="https://github.com/cpvalente/ontime/assets/58940073/3fd93e3b-0cb0-4944-9750-82fde7cd2adc">

Reply:
Thats right, I guess in an object property it would never be undeclared, good suggestion